src/toxcore/dht/client.rs

```diff
@@ -135,25 +135,51 @@ impl Client {
                 IfAddr::V4(ref addr) => addr.broadcast,
                 _ => None,
         })
-        .map(|ipv4|
-            IpAddr::V4(ipv4)
-        ).collect()
+        .map(|addr|
+            IpAddr::V4(addr)
+        )
+        .collect()
+    }
+    // get broadcast addresses for host's network interfaces
+    fn get_ipv6_broadcast_addrs() -> Vec<IpAddr> {
```

Comment:
Is it possible that IPv6 has broadcast address? According to wiki:
> Internet Protocol version 6 (IPv6), the next generation Internet Protocol, does not implement the method of broadcast, and therefore does not define broadcast addresses.

Reply:
Yes, but FF02::01 is used for broadcasting.

Reply:
FF02::01 is a multicast address for all nodes in local network.
They just set netmask in broadcast field in case of ipv6:
https://github.com/maidsafe/get_if_addrs/blob/master/src/lib.rs#L259
It seems to me it's not correct to use it as broadcast address.

Reply:
Yes. we are saying same thing.
Lan Discovery is for local network.

Reply:
this function is given by crate named "get_if_addrs".
I guess that this function may return FF02::01.

Reply:
See sorce code: https://github.com/maidsafe/get_if_addrs/blob/master/src/lib.rs#L259
It will return netmasks (I suppose ffff:ffff:ffff:ffff::).

Reply:
Thanks for your comments.
I will revise code on this.